test/unit/reducers/projects.js

```diff
@@ -164,20 +172,44 @@ tap(initProjects({1: false}), projects =>
   )),
 );
 
+tap(initProjects({1: false}), projects =>
+  test('hideComponent', reducerTest(
+    reducer,
+    projects,
+    partial(hideComponent, '1', 'output', now),
+    projects.update('1', projectIn =>
+      projectIn.set('hiddenUIComponents', new Immutable.Set(['output'])).
+        set('updatedAt', now),
+    ),
+  )),
+);
+
+tap(initProjects({1: true}), projects =>
+  test('unhideComponent', reducerTest(
+    reducer,
+    projects,
+    partial(unhideComponent, '1', 'output', now),
+    projects.update('1', projectIn =>
+      projectIn.set('hiddenUIComponents', new Immutable.Set()).
+        set('updatedAt', now),
+    ),
+  )),
+);
+
 function initProjects(map = {}) {
   return reduce(map, (projectsIn, modified, key) => {
-    const projects = reducer(projectsIn, projectCreated(key));
+    let projects = reducer(projectsIn, projectCreated(key));
     if (modified) {
-      return reducer(
-        projects,
-        updateProjectSource(key, 'css', '', now),
-      );
+      projects = reducer(projects, updateProjectSource(key, 'css', '', now));
+      projects = reducer(projects, hideComponent(key, 'output', now));
```

Comment:
I wanted an easy way to test `unhideComponent()`, so I needed the stub project's "before" state to have a currently hidden component. I just co-opted the `modified` flag to add a hidden component when the stub project is being created. Is there a better stub to use?